src/Persistence/RdsQueue.php

```diff
@@ -90,25 +89,38 @@ public function createTask($action, $parameters, $repeatedly = false, $label = n
      * @param $stateId
      * @param string $report
      */
-    public function updateTaskStatus($taskId, $stateId, $report = '' )
+    public function updateTaskStatus($taskId, $stateId)
     {
-        $task = $this->getTask($taskId);
-        $task->setReport($report);
-        $task->setStatus($stateId);
-
         $platform = $this->getPersistence()->getPlatForm();
         $statement = 'UPDATE '.self::QUEUE_TABLE_NAME.' SET '.
             self::QUEUE_STATUS.' = ?, '.
-            self::QUEUE_UPDATED.' = ?, '.
-            self::QUEUE_REPORT.' = ?, '.
-            self::QUEUE_TASK.' = ? '.
+            self::QUEUE_UPDATED.' = ? '.
             'WHERE '.self::QUEUE_ID.' = ?';
 
         $this->getPersistence()->exec($statement, [
             $stateId,
             $platform->getNowExpression(),
+            $taskId
+        ]);
+    }
+
+
+    /**
+     * @param string $taskId
+     * @param $stateId
+     * @param string $report
+     */
+    public function updateTaskReport($taskId, $report)
+    {
+        $platform = $this->getPersistence()->getPlatForm();
+        $statement = 'UPDATE '.self::QUEUE_TABLE_NAME.' SET '.
+            self::QUEUE_UPDATED.' = ?, '.
+            self::QUEUE_REPORT.' = ? '.
+            'WHERE '.self::QUEUE_ID.' = ?';
+
+        $this->getPersistence()->exec($statement, [
+            $platform->getNowExpression(),
             json_encode($report),
-            json_encode($task),
```

Comment:
You have to store fully serialized task there. There are a few duplicated data (for example status stored in serialized json and in appropriate table column) but data in table columns is used only for search purposes. Otherwise you will have to add column to task if you added new task property (for example if you will decide extend JsonTask class).